From: Stephan Linz <linz@li-pro.net>
Date: Sat, 28 Oct 2017 07:57:52 +0200
Subject: declare phony and forced rules for special make targets

In terms of Make, a phony target is simply a target that is always
out-of-date, so whenever you ask 'make <phony_target>', it will run,
independent from the state of the file system. Some common make
targets that are often phony are: all, install, clean.

https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html

If a rule has no prerequisites or recipe, and the target of the rule
is a nonexistent file, then make imagines this target to have been
updated whenever its rule is run. This implies that all targets
depending on this one will always have their recipe run. The target
simz80.c is such kind of rule to ensure it will generated by simz80.pl
independently fom the state of the file system.

https://www.gnu.org/software/make/manual/html_node/Force-Targets.html

Signed-off-by: Stephan Linz <linz@li-pro.net>
---
 Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile b/Makefile
index 83d609c..d3b3e42 100644
--- a/Makefile
+++ b/Makefile
@@ -64,6 +64,7 @@ all:		yaze cdm
 yaze:		$(YAZE_OBJS)
 		$(CC) $(CFLAGS) $(YAZE_OBJS) $(LIBS) -o $@
 
+simz80.c:	.FORCE
 simz80.c:	simz80.pl
 		rm -f simz80.c
 		perl -w simz80.pl >simz80.c
@@ -104,3 +105,6 @@ clean:;		rm -f *.o *~ core yaze cdm
 yaze.o:		yaze.c simz80.h yaze.h
 bios.o:		bios.c simz80.h yaze.h
 simz80.o:	simz80.c simz80.h
+
+.PHONY: .FORCE all clean install tar
+.FORCE:
